compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -963,6 +965,19 @@ public void visit(BLangArrayType arrayType) {
 
     @Override
     public void visit(BLangUserDefinedType userDefinedType) {
+        BType type = userDefinedType.getBType();
+        if (type == null) {
```

Comment:
When can the type be null?

Reply:
Updated. Earlier, In the desugaring phase, we created a user-defined type for the local record.

Reply:
Move this logic to record type node visitor.